hello-world/tutorial/index.md

```diff
@@ -0,0 +1,471 @@
+	switch dispatchedAck := ack.Response.(type) {
+	case *channeltypes.Acknowledgement_Error:
+		// We will not treat acknowledgment error in this tutorial
+		return nil
+	case *channeltypes.Acknowledgement_Result:
+		// Decode the packet acknowledgment
+		var packetAck types.IbcPostPacketAck
+		err := packetAck.Unmarshal(dispatchedAck.Result)
+		if err != nil {
+			// The counter-party module doesn't implement the correct acknowledgment format
+			return errors.New("cannot unmarshal acknowledgment")
+		}
+
+		k.AppendSentPost(
+			ctx,
+			data.Creator,
+			packetAck.PostID,
+			data.Title,
+			packet.DestinationPort+"-"+packet.DestinationChannel,
+		)
+
+		return nil
+	default:
+		// The counter-party module doesn't implement the correct acknowledgment format
+		return errors.New("invalid acknowledgment format")
+	}
+}
+```
+
+### Storing information about timed out packet
+
+In `timedoutPost` posts we store posts that have not been received by target chains. It follows the same format as `sentPost` 
+
+```go
+// OnTimeoutIbcPostPacket responds to the case where a packet has not been transmitted because of a timeout
+func (k Keeper) OnTimeoutIbcPostPacket(ctx sdk.Context, packet channeltypes.Packet, data types.IbcPostPacketData) error {
+
+	k.AppendTimedoutPost(
+		ctx,
+		data.Creator,
+		data.Title,
+		packet.DestinationPort+"-"+packet.DestinationChannel,
+	)
+
+	return nil
+}
+```
+
+That is our basic `blog` module setup.
+The blockchain is now ready!
+
+Let's spin it up and send a blog post from one blockchain app to the other.
+We will need various terminal windows for this.
+
+## IBC module tests
+
+### Starting the chains
+
+To test the IBC module, we will serve with starport two blockchain networks on the same machine. Both blockchains uses the same source code but with a different chain ID
+
+We will have a blockchain network named `earth` and the other name `mars`
+
+We need the two following files in the project directory:
+
+`planet/earth.yml`
+
+```yaml
+accounts:
+  - name: alice
+    coins: ["1000token", "100000000stake"]
+  - name: bob
+    coins: ["500token"]
+validator:
+  name: alice
+  staked: "100000000stake"
+faucet:
+  name: bob
+  coins: ["5token"]
+genesis:
+  chain_id: "earth"
+init:
+  home: "$HOME/.earth"
+```
+
+`planet/mars.yml`
+
+```yaml
+accounts:
+  - name: alice
+    coins: ["1000token", "1000000000stake"]
+  - name: bob
+    coins: ["500token"]
+validator:
+  name: alice
+  staked: "100000000stake"
+faucet:
+  port: 4501
+  name: bob
+  coins:
+    - "5token"
+servers:
+  rpc-address: "0.0.0.0:26659"
+  p2p-address: "0.0.0.0:26658"
+  prof-address: "0.0.0.0:6061"
+  grpc-address: "0.0.0.0:9091"
+  api-address: "0.0.0.0:1318"
+  frontend-address: "0.0.0.0:8081"
+  dev-ui-address: "0.0.0.0:12346"
+genesis:
+  chain_id: "mars"
+init:
+  home: "$HOME/.mars"
+```
+
+We can then run the two blockchain networks with the following commands on two separate terminals:
```

Comment:
```suggestion
Use a  different terminal window for each blockchain. Run the following commands for each blockchain:
```

Reply:
should we agree to always call a blockchain a blockchain? It seems as if the term "blockchain network" is not used consistently... teach me

Reply:
A blockchain is a blockchain, agreed 👍